frontend/src/Editor/QueryManager/QueryEditors/TooljetDatabase/AggregateUI/index.jsx

```diff
@@ -0,0 +1,437 @@
+          delete currentAggregates[aggregateKey];
+          handleChange('aggregates', currentAggregates);
+          toast.success('Aggregate function deleted successfully!');
+          return;
+        } catch (error) {
+          return toast.error('Could not delete aggregate function. Please try again!');
+        }
+      }
+    }
+  };
+
+  const executeAggregateDeletion = (aggregateKey) => {
+    try {
+      const currentAggregates = { ...(operationDetails?.aggregates || {}) };
+      delete currentAggregates[aggregateKey];
+      const currentGroupBy = { ...(operationDetails?.group_by || {}) };
+      delete currentGroupBy?.[selectedTableId];
+
+      handleChange('group_by', currentGroupBy);
+      handleChange('aggregates', currentAggregates);
+      toast.success('Aggregate function deleted successfully!');
+      return;
+    } catch (error) {
+      return toast.error('Could not delete aggregate function. Please try again!');
+    }
+  };
+
+  const handleGroupByChange = (selectedTableId, value) => {
+    const currentGroupBy = { ...(operationDetails?.group_by || {}) };
+    const validValueData = value?.reduce((acc, val) => {
+      if (typeof val === 'object' && !acc.some((option) => option === val.value)) {
+        acc.push(val.value);
+      } else if (typeof val !== 'object' && !acc.some((option) => option === val.value)) {
+        acc.push(val);
+      }
+      return acc;
+    }, []);
+    const updatedGroupBy = {
+      ...currentGroupBy,
+      [selectedTableId]: validValueData,
+    };
+    handleChange('group_by', updatedGroupBy);
+  };
+
+  const columnAccessorsOptions = useMemo(() => {
+    return columns.map((column) => {
+      return {
+        label: column.accessor,
+        value: column.accessor,
+      };
+    });
+  }, [columns]);
+
+  const disableGroupBy = () => {
+    const currentAggregates = { ...(operationDetails?.aggregates || {}) };
+    const isAnyAggregateTruthyValue = isEmpty(currentAggregates)
+      ? false
+      : Object.values(currentAggregates).some((aggregate) => {
+          if (aggregate.aggFx && aggregate.column) {
+            return true;
+          } else {
+            return false;
+          }
+        });
+    return !isAnyAggregateTruthyValue;
+  };
+  const getTableName = (id) => {
+    return tables.find((table) => table.table_id === id)?.table_name;
+  };
+
+  const tableListOptions = useMemo(() => {
+    const tableList = [];
+
+    const tableSet = new Set();
+    (joinOptions || []).forEach((join) => {
+      const { table, conditions } = join;
+      tableSet.add(table);
+      conditions?.conditionsList?.forEach((condition) => {
+        const { leftField, rightField } = condition;
+        if (leftField?.table) {
+          tableSet.add(leftField?.table);
+        }
+        if (rightField?.table) {
+          tableSet.add(rightField?.table);
+        }
+      });
+    });
+
+    const tablesDetails = [...tableSet];
+
+    tablesDetails.forEach((tableId) => {
```

Comment:
We cannot implement caching here because we depend on table schema to fetch all columns available in that table. If that particular schema gets updated (columns), we may have outdated columns available due to catching.